src/Graphics/src/Graphics/ICanvas.cs

```diff
@@ -62,37 +244,117 @@ public interface ICanvas
+		/// </summary>
+		/// <remarks>Can cause starting coordinates to move when an object is made larger.</remarks>
+		/// <param name="sx">Value for horizontal scaling.</param>
+		/// <param name="sy">Value for vertical scaling.</param>
 		public void Scale(float sx, float sy);
 
+		/// <summary>
+		/// Shifts a graphical object in horizontal and vertical directions.
+		/// </summary>
+		/// <param name="tx">Horizontal shift. Negative values move the object to the left, while positive values move it to the right.</param>
+		/// <param name="ty">Vertical shift. Negative values move the object down, while positive values move it up.</param>
 		public void Translate(float tx, float ty);
 
+		/// <summary>
+		/// Applies transformation specified by <paramref name="transform"/> to a graphical object.
+		/// </summary>
+		/// <param name="transform">Affine transformation matrix.</param>
 		public void ConcatenateTransform(Matrix3x2 transform);
 
+		/// <summary>
+		/// Saves the current graphics state.
+		/// </summary>
 		public void SaveState();
 
+		/// <summary>
+		/// Restores the graphics state to the most recently saved state.
+		/// </summary>
+		/// <returns><see langword="true"/> if the restore was succesful, <see langword="false"/> otherwise.</returns>
 		public bool RestoreState();
 
+		/// <summary>
+		/// Resets the graphics state to its default values.
+		/// </summary>
 		public void ResetState();
 
+		/// <summary>
+		/// Adds a shadow to a graphical object.
+		/// </summary>
+		/// <param name="offset">Represents the position of a light source that creates the shadow.</param>
+		/// <param name="blur">Amount of blur to apply to the shadow.</param>
+		/// <param name="color">Color of the shadow.</param>
 		public void SetShadow(SizeF offset, float blur, Color color);
 
+		/// <summary>
+		/// Sets <paramref name="paint"/> as the fill of a graphical object.
+		/// </summary>
+		/// <param name="paint">Paint to set as fill</param>
+		/// <param name="rectangle">Rectangle to apply a gradient on if the <paramref name="paint"/> supports it.</param>
 		public void SetFillPaint(Paint paint, RectF rectangle);
 
+		/// <summary>
+		/// Draws an image onto the canvas.
+		/// </summary>
+		/// <param name="image">Image to display</param>
+		/// <param name="x">Top left corner <c>x</c> coordinate.</param>
+		/// <param name="y">Top left corner <c>y</c> coordinate.</param>
+		/// <param name="width">Width of the image.</param>
+		/// <param name="height">Height of the image.</param>
 		public void DrawImage(IImage image, float x, float y, float width, float height);
 
+		/// <summary>
+		/// Calculates the are a string would occupy if drawn on the canvas.
+		/// </summary>
+		/// <param name="value">Text to calculate the size on.</param>
+		/// <param name="font"></param>
+		/// <param name="fontSize"></param>
+		/// <returns>The size of </returns>
 		public SizeF GetStringSize(string value, IFont font, float fontSize);
```

Comment:
@mattleibow please confirm that I interpreted this method correctly.

Reply:
Correct. This "measures" the string.